server/user.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var childProcess = require('child_process');
+var readline = require('readline');
+var mkdirp = require('mkdirp');
+
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGHUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Name of user information file.
+ * @type {string}
+ */
+user.userInfoFile = 'userInfo.txt';
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user indormation file.
+ * @param {!Object} userInfo User Information
+ * @return {!Object{
+ *   success: boolean,
+ *   errorMessage: {=string}
+ * }}
+ */
+user.signUp = function(userPath, userInfo) {
+  var checkDuplicate = {
+    duplicated: false,
+    element: []
+  };
+  var lines = fs.readFileSync(userPath + user.userInfoFile).toString()
+    .split('\n');
+  for (var i = 0; i < lines.length; i++) {
+    var parts = lines[i].split(/[\t\s]+/);
+    if (parts.length == 0) {
+      continue;
+    }
+    if (parts[0] == userInfo.email || parts[1] == userInfo.username) {
```

Comment:
You need to also validate the username and email.
Username should not contain special characters.
Email should be valid email string.